### ISSUE TITLE
Support Inline Images Protocol

### ISSUE THREAD

**Issue by Niek** [cronvel/terminal-kit#124]
It would be great if this library would support the Iterm Inline Images Protocol. See https://www.iterm2.com/documentation-images.html for details.

See for example: https://github.com/sindresorhus/ansi-escapes/blob/dcd5ef3c124079c693bef1725044d91b0bca6e7f/index.js#L112-L128

**Comment by cronvel** [comment]
@Niek I already heard of the Iterm2 image capabilities, but I don't have MacOS, so I either need a dedicated MacOS maintainer for this project, or a way to test that on Linux. Last time I checked, there wasn't any iTerm2 clone on Linux with such capabilities... 

I could add the feature blindly, but I usually hate that...

**Comment by Niek** [comment]
@cronvel good point! At least wezterm (Linux/Windows/macOS) seems to have added the same support: https://wezfurlong.org/wezterm/imgcat.html

Maybe there are more terminals out there that support it, but I'm not aware of it yet.

**Comment by Digicrat** [comment]
I do wish that more terminal emulators supported images ... though it looks like that may be on the cusp of changing.

[Sixel](https://saitoha.github.io/libsixel/) is another image format, similar to iterm2s (and supposedly supported by as well), that seems to be more widely supported.  

It looks like libSixel support isn't quite mainstream, but it seems to work in iTerm2 and "mlterm" (which is in the Ubuntu package manager and just worked for me).  Support is also pending for it in alacritty, and xterm apparently supports it if built from source with the appropriate option.

I haven't delved too deeply into it, but there is a control sequence that allows for detection of sixel support (and a node package supports-sixel to do this).  Ideally then, the existing terminal-kit drawImage could be extended to seamlessly switch to using the sixel format if supported by the terminal, and fallback to the current low-res implementation if not.

**Comment by cronvel** [comment]
@Digicrat If *Sixel* is going to be the cross platform standard, then it's the way to go. However, I'm still waiting for a wider adoption at this moment.

Also, what is the capability of Sixel? It seems to me that the number of colors is quite low and use dithering to simulate more colors.

While I would usually prefer a more standard solution, if it's made obsolete by iTerm2 image protocol in near future, I would definitively prefer the later.

**Comment by Niek** [comment]
Sixel looks interesting but in my limited testing it has 2 issues:
* Lower quality (limited colors) compared to Inline Images Protocol
* High CPU usage when displaying a simple GIF
<img width="729" alt="Screen Shot 2021-04-28 at 14 28 51" src="https://user-images.githubusercontent.com/213140/116403812-500e2b80-a82e-11eb-8297-32193f5f15c2.png">

On the other hand, with Inline Images Protocol it takes much longer to render the initial frame because of the base64 encoding. Also iTerm takes quite some CPU usage too:
<img width="633" alt="Screen Shot 2021-04-28 at 14 29 29" src="https://user-images.githubusercontent.com/213140/116403916-6caa6380-a82e-11eb-800e-b9a5ee8f5eae.png">

**Comment by jawspeak** [comment]
It looks like the image protocol (let's say iterm2's) is actually very simple.

It seems you can use terminal-image npm package to generate the escape characters and base64 encoded image to render directly in terminal-kit. 

```
        const imgPath = 'my/image-test.jpeg';
        term.noFormat(await terminalImage.file(imgPath, { width: 25 }));
```
